automation-tests/test/utilities.ts

```diff
@@ -0,0 +1,337 @@
+  if (!terminalView) {
+    throw new Error('In executeCommand(), the terminal view returned from getTerminalView() was null (or undefined)');
+  }
+
+  await terminalView.executeCommand(command);
+
+  return terminalView;
+}
+
+async function getTerminalView(workbench: Workbench): Promise<TerminalView> {
+  const bottomBar = await workbench.getBottomBar();
+  const terminalView = await bottomBar.openTerminalView();
+  await utilities.pause(5);
+
+  return terminalView;
+}
+
+async function getTerminalViewText(terminalView: TerminalView, seconds: number): Promise<string> {
+  for (let i=0; i<seconds; i++) {
+    await pause(1);
+
+    // const terminalText = await terminalView.getText();
+    // terminalView.getText() no longer works
+
+    const workbench = await (await browser.getWorkbench()).wait();
+    await browser.keys([CMD_KEY, 'a', 'c']);
+    // Should be able to use Keys.Ctrl, but Keys is not exported from webdriverio
+    // See https://webdriver.io/docs/api/browser/keys/
+    const terminalText = await clipboard.read();
+
+    if (terminalText && terminalText !== '') {
+      return terminalText
+    }
+  }
+
+  throw new Error('Exceeded time limit - text in the terminal was not found');
+}
+
+async function getFilteredVisibleTreeViewItems(workbench: Workbench, projectName: string, searchString: string): Promise<ViewItem[]> {
+  const sidebar = workbench.getSideBar();
+  const treeViewSection = await sidebar.getContent().getSection(projectName);
+  await treeViewSection.expand();
+
+  // Warning, we can only retrieve the items which are visible.
+  const visibleItems = (await treeViewSection.getVisibleItems()) as DefaultTreeItem[];
+  const filteredItems = (await visibleItems.reduce(async (previousPromise: Promise<ViewItem[]>, currentItem: ViewItem) => {
+    const results = await previousPromise;
+    const label = await (currentItem as TreeItem).getLabel();
+    if (label.startsWith(searchString)) {
+      results.push(currentItem);
+    }
+
+    return results;
+  }, Promise.resolve([])) as ViewItem[]);
+
+  return filteredItems;
+}
+
+async function getFilteredVisibleTreeViewItemLabels(workbench: Workbench, projectName: string, searchString: string): Promise<string[]> {
+  const sidebar = workbench.getSideBar();
+  const treeViewSection = await sidebar.getContent().getSection(projectName);
+  await treeViewSection.expand();
+
+  // Warning, we can only retrieve the items which are visible.
+  const visibleItems = (await treeViewSection.getVisibleItems()) as DefaultTreeItem[];
+  const filteredItems = (await visibleItems.reduce(async (previousPromise: Promise<string[]>, currentItem: ViewItem) => {
+    const results = await previousPromise;
+    const label = await (currentItem as TreeItem).getLabel();
+    if (label.startsWith(searchString)) {
+      results.push(label);
+    }
+
+    return results;
+  }, Promise.resolve([])) as string[]);
+
+  return filteredItems;
+}
+
+function currentUserName(): string {
```

Comment:
nit: I might make this `currentOsUsername` just b/c username is already assumed to be the SF username.